lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -1,9 +1,61 @@
 """
 Progress Tab Serializers
 """
-
 from rest_framework import serializers
 from lms.djangoapps.course_home_api.outline.v1.serializers import CourseBlockSerializer
+from rest_framework.reverse import reverse
+
+
+class ProblemScoresSerializer(serializers.Serializer):
+    earned = serializers.FloatField()
+    possible = serializers.FloatField()
+
+
+class GradedTotalSerializer(serializers.Serializer):
+    first_attempted = serializers.CharField()
+    possible = serializers.FloatField()
+    earned = serializers.FloatField()
+    graded = serializers.BooleanField()
+
+
+class SubsectionSerializer(serializers.Serializer):
+    display_name = serializers.CharField()
+    url = serializers.SerializerMethodField()
+    percent_graded = serializers.FloatField()
+    format = serializers.CharField()
+    due = serializers.CharField()
+    # TODO: override serializer
+    show_correctness = serializers.CharField()
+    graded = serializers.BooleanField()
+    graded_total = GradedTotalSerializer()
+    problem_scores = serializers.SerializerMethodField()
+    show_grades = serializers.SerializerMethodField()
+
+    def get_url(self, subsection):
+        relative_path = reverse('jump_to', args=[self.context['course_key'], subsection.location])
+        request = self.context['request']
+        return request.build_absolute_uri(relative_path)
+
+    def get_problem_scores(self, subsection):
+        problem_scores = [
+            {
+                'earned': score.earned,
+                'possible': score.possible,
+            }
+            for score in subsection.problem_scores.values()
+        ]
+        return problem_scores
+
+    def get_show_grades(self, subsection):
+        return subsection.show_grades(self.context['staff_access'])
+
+
+class ChapterSerializer(serializers.Serializer):
+    """
+    Serializer for chapters in coursewaresummary
+    """
+    display_name = serializers.CharField()
+    sections = SubsectionSerializer(many=True) # this sections actually refers to subsections in the chapter
```

Comment:
So if you're feeling up to it (it will require updating the frontend PR as well), I think you can change this to 
```
subsections = SubsectionSerializer(source='sections', many=True)
```
And that will rename the field in the serializer to subsections, but still source all the information from the sections field in the chapter.